docs/architecture/high-performance-scala.md

```diff
@@ -90,9 +100,45 @@ readability. They've been introduced merely to overcome shortcomings in
 certain boxing-behaviors in the scala compiler and understanding their
 behavior isn't necessary to read/understand the GeoTrellis codebase.
 
+# Micro-Optimizations
+------
 
+## Loops ##
 
-# Macros
------
-# Spark
-------
\ No newline at end of file
+A commonly-seen feature throughout the codebase is the use of `cfor`- or `while`-loops where it would seem that an ordinary `for`-loop would be sufficient.
+We avoid using "`for`-loops" in most cases because they are more than just loops in Scala.
```

Comment:
Yeah, I was trying to decide how to approach it, as well, that is why I put it in quotation marks.